GVFS/GVFS.UnitTests/Common/LibGit2RepoPoolTests.cs

```diff
@@ -0,0 +1,174 @@
+﻿using GVFS.Common;
+using GVFS.Common.Git;
+using GVFS.Tests.Should;
+using GVFS.UnitTests.Mock.Common;
+using NUnit.Framework;
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace GVFS.UnitTests.Common
+{
+    [TestFixture]
+    public class LibGit2RepoPoolTests
+    {
+        private readonly TimeSpan disposalPeriod = TimeSpan.FromMilliseconds(1);

```

Comment:
What happens if the disposal interval is zero/instantaneous? Should we be checking for a non-zero interval in the constructor of `LibGit2RepoPool`?